src/core/MOM_CoriolisAdv.F90

```diff
@@ -428,45 +431,48 @@ subroutine CorAdCalc(u, v, h, uh, vh, CAu, CAv, OBC, AD, G, GV, US, CS)
       endif
     enddo ; endif
 
+    if (CS%no_slip) then
+      do J=Jsq-1,Jeq+1 ; do I=Isq-1,Ieq+1
+        rel_vort(I,J) = (2.0 - G%mask2dBu(I,J)) * (dvdx(I,J) - dudy(I,J)) * G%IareaBu(I,J)
+      enddo ; enddo
+    else
+      do J=Jsq-1,Jeq+1 ; do I=Isq-1,Ieq+1
+        rel_vort(I,J) = G%mask2dBu(I,J) * (dvdx(I,J) - dudy(I,J)) * G%IareaBu(I,J)
+      enddo ; enddo
+    endif
+
     do J=Jsq-1,Jeq+1 ; do I=Isq-1,Ieq+1
-      if (CS%no_slip ) then
-        relative_vorticity = (2.0-G%mask2dBu(I,J)) * (dvdx(I,J) - dudy(I,J)) * G%IareaBu(I,J)
-      else
-        relative_vorticity = G%mask2dBu(I,J) * (dvdx(I,J) - dudy(I,J)) * G%IareaBu(I,J)
-      endif
-      absolute_vorticity = G%CoriolisBu(I,J) + relative_vorticity
-      Ih = 0.0
-      if (Area_q(i,j) > 0.0) then
-        hArea_q = (hArea_u(I,j) + hArea_u(I,j+1)) + (hArea_v(i,J) + hArea_v(i+1,J))
-        Ih = Area_q(i,j) / (hArea_q + h_neglect*Area_q(i,j))
-      endif
-      q(I,J) = absolute_vorticity * Ih
-      abs_vort(I,J) = absolute_vorticity
-      Ih_q(I,J) = Ih
-
-      if (CS%bound_Coriolis) then
-        fv1 = absolute_vorticity * v(i+1,J,k)
-        fv2 = absolute_vorticity * v(i,J,k)
-        fu1 = -absolute_vorticity * u(I,j+1,k)
-        fu2 = -absolute_vorticity * u(I,j,k)
-        if (fv1 > fv2) then
-          max_fvq(I,J) = fv1 ; min_fvq(I,J) = fv2
-        else
-          max_fvq(I,J) = fv2 ; min_fvq(I,J) = fv1
-        endif
-        if (fu1 > fu2) then
-          max_fuq(I,J) = fu1 ; min_fuq(I,J) = fu2
-        else
-          max_fuq(I,J) = fu2 ; min_fuq(I,J) = fu1
-        endif
-      endif
+      abs_vort(I,J) = G%CoriolisBu(I,J) + rel_vort(I,J)
+    enddo ; enddo
 
-      if (CS%id_rv > 0) RV(I,J,k) = relative_vorticity
-      if (CS%id_PV > 0) PV(I,J,k) = q(I,J)
-      if (associated(AD%rv_x_v) .or. associated(AD%rv_x_u)) &
-        q2(I,J) = relative_vorticity * Ih
+    do J=Jsq-1,Jeq+1 ; do I=Isq-1,Ieq+1
+      hArea_q = (hArea_u(I,j) + hArea_u(I,j+1)) + (hArea_v(i,J) + hArea_v(i+1,J))
+      Ih_q(I,J) = merge( &
```

Comment:
Although merge is a standard F90 intrinsic, it seems obscure to me and does not have a direct functional tie to the English language (unlike where), and is not used elsewhere in the MOM6 code. I would suggest that we add a comment here like "! x = merge(a, b, test) is equivalent to 'if (test) then ; x=a ; else ; x=b ; endif`, to help non-proficient F90 programmers like me to understand what this is doing.  Alternately, should we search through the rest of the code for things like this explicit logical assignment and replace them all with merge() calls, since they are so much faster.

It is very surprising to me that replacing the straightforward explicit logic syntax with a merge is faster because it seems like compilers should be able to see through this! What are compilers good for?

Another curious point here - I would have written an explicit 5-command conditional assignment in 1 line with semicolons, whereas this is a 1-command conditional assignment via a merge written in 5 lines.  To each his own, I guess...

Reply:
I would suggest not replacing isolated if-blocks with `merge` statements at this time.  Although it had a positive effect here, often they do much of anything.

As for the etymology, usually it's used to "merge" two arrays based on a third logical array.  Of course, for a scalar that merge basically just reduces to an if-block.  My guess is that somehow the intention to merge is getting interpreted as a vectorized operation.

Reply:
Replacing `h_neglect * Area_q` with a new `vol_neglect` will allow `Area_q` to act as its own mask, and we can avoid the `merge` altogether.  I will update the patch soon.